vortex-scalar/src/primitive.rs

```diff
@@ -103,6 +105,27 @@ impl<'a> PrimitiveScalar<'a> {
             }?)),
         }
     }
+
+    pub fn subtract(&self, other: &PrimitiveScalar) -> VortexResult<Self> {
```

Comment:
Could you call this `checked_sub(&self, other: &PrimitiveScalar) -> VortexResult<Option<Self>>` and then use num-traits CheckedSub to ensure the subtraction doesn't overflow. Otherwise this function can panic.

Otherwise looks good to me!